src/compiler/utilities.ts

```diff
@@ -2656,6 +2656,26 @@ namespace ts {
         return result || emptyArray;
     }
 
+    function filterOwnedJSDocTags(hostNode: Node, jsDoc: JSDoc | JSDocTag) {
+        if (isJSDoc(jsDoc)) {
+            const ownedTags = filter(jsDoc.tags, tag => ownsJSDocTag(hostNode, tag));
+            return jsDoc.tags === ownedTags ? [jsDoc] : ownedTags;

```

Comment:
why isn't the true branch `ownedTags` too?

Reply:
`ts.filter` returns the same array if all elements are included. If `ownedTags` is the same array as `jsDoc.tags`, then we just return `jsDoc` (since that's what we included in the result before this). Otherwise, we return the filtered list of tags.

This may be overkill, but the reason why we return both `JSDoc` and `JSDocTag` entries isn't clear so I felt safer returning the original node if no filtering occurred.